apps/server/src/infra/sync/tsp/tsp-sync.strategy.ts

```diff
@@ -76,4 +93,31 @@ export class TspSyncStrategy extends SyncStrategy {
 
 		return scSchools.filter((scSchool) => scSchool != null).map((scSchool) => scSchool.school);
 	}
+
+	private async syncData(system: System, schools: School[]): Promise<void> {
+		const tspTeachers = await this.tspSyncService.fetchTspTeachers(system, this.schoolDataDaysToFetch);
+		const tspStudents = await this.tspSyncService.fetchTspStudents(system, this.schoolDataDaysToFetch);
+		const tspClasses = await this.tspSyncService.fetchTspClasses(system, this.schoolDataDaysToFetch);
+		this.logger.info(
+			new TspDataFetchedLoggable(tspTeachers.length, tspStudents.length, tspClasses.length, this.schoolDataDaysToFetch)
+		);
+
+		const oauthDataDtos = this.tspSyncService.mapTspDataToOauthData(
+			system,
+			schools,
+			tspTeachers,
+			tspStudents,
+			tspClasses
+		);
+
+		this.logger.info(new TspSyncingUsersLoggable(oauthDataDtos.length));
+
+		const dataPromises = oauthDataDtos.map((oauthDataDto) =>
+			this.dataLimit(() => this.provisioningService.provisionData(oauthDataDto))
+		);
+
+		const results = await Promise.all(dataPromises);
+
```

Comment:
Use Promise.allSettled so the sync does not cancle because of one error.

Reply:
done